File: pages/agent_page.py
import json
from playwright.sync_api import Page
from .base_page import BasePage


class AgentPage(BasePage):
    """Locators and methods for the Agent Page"""
    URL = f"{BasePage.BASE_URL}concierge-team"
    TITLE = "Find an Agent | ComeHome"

    # Locators
    def __init__(self, page: Page):
        super().__init__(page)
        self.page = page

        # Load in common selectors from json file
        with open(BasePage.SELECTORS_FILE_PATH, mode="r", encoding="utf-8") as json_data:
            selectors = json.load(json_data)['agent_page']

        # Top Section
        self.topSection = self.page.locator(selectors['topSection'])

        # Sub Section
        self.subSection = self.page.locator(selectors['subSection'])
    
    # Methods
    def goto(self):
        self.page.goto(self.URL)

File: pages/agent_page.js
import { BasePage} from './base_page'
import { agent_page } from '../common/selectors.json';


exports.AgentPage = class AgentPage extends BasePage {
    // Locators
    constructor(page) {
        super(page);
        this.page = page;

        this.topSection = page.locator(agent_page.topSection);
        this.subSection = page.locator(agent_page.subSection);
    }
    
    // Methods
    async gotoAgentPage() {
        await this.page.goto('/concierge-team');
    }
}
